Automattic/woocommerce-payments
# Instant Deposits: Show amount on a notice and button label

Fixes #8220 

The PR implements a notice, updated button text and a tooltip for `Instant deposits` on the `Overview` page. The UI update will clearly inform the merchant of the exact amount available for instant deposit. 

### Screenshots

#### With notice
<img src="https://github.com/Automattic/woocommerce-payments/assets/4162931/8f924962-0185-4e34-b492-18fe167fec2d" width="500px" />

#### After notice is dismissed
<img src="https://github.com/Automattic/woocommerce-payments/assets/4162931/686394e0-8576-4c29-b73f-c41de3f842a9" width="500px" />

#### Tooltip
<img src="https://github.com/Automattic/woocommerce-payments/assets/4162931/72118702-df6f-4d90-b285-f7b98aaa60ce" width="500px" />


#### Testing instructions
* On your test account use the instructions from the following links to have it ready with some instant deposit eligible balance
  * PdjTHR-2XG-p2 - Deposits Cheatsheet 
  * pdjTHR-3oY-p2 - - Demystifying Instant deposits 
  * You may sync your local test site to `acct_1LqHU4QocNIKfAAZ` . This account is approved for instant deposit and should show you the notice and button. 
* Browse to the Payments page. You should see a view similar to the screenshot `With notice` . Note that there is no tooltip icon visible
* The button and the notice should both indicate the correct values for the available amount for instant deposit. The notice should also show the correct fee % ( 1.5% )
* Dismiss the notice
* You will now see tooltip icon 
* On clicking, you should see the message, similar to what you see in the screenshot `Tooltip` 
* `Learn more` link should lead to - https://woocommerce.com/document/woopayments/deposits/instant-deposits/
* If you reload the page, the notice should not appear again. This should persist until you delete the option `wcpay_instant_deposit_notice_dismissed` from the client test site DB. 

#### Points of discussion / decision needed
These are minor changes and can be easily actioned depending on the outcome of the discussion.

* ✅ - Text updated via f93fecb -  ~Whether we need a [change in the text ](https://github.com/Automattic/woocommerce-payments/issues/8220#issuecomment-2175399487) to prevent confusion.~
* ✅  Test to be removed ~[Opinion needed](https://github.com/Automattic/woocommerce-payments/pull/8970#issuecomment-2175432057) - Not directly related, and minor test refactor~
* ✅  - Done - ~How to persist notice dismissal better - 1718694806082599-slack-C02BW3Z8SHK~

### Dev notes
* Figma design : https://www.figma.com/design/aDFNTeogq2cXSr4Nqeni8g/Instant-Deposits?node-id=768-76519&m=dev
* Used the pattern on this PR to persist the dismissed notice state using user settings. Props to @Jinksi  and @shendy-a8c 🙌 
-------------------


**Post merge**
